File: time.c
#include <stdio.h>
#include "vecchia.h"
#include <omp.h>

int main(){

    FILE *obs;
    FILE *nn;
    FILE *l;

    obs  = fopen("data/y.txt", "r");
    nn   = fopen("data/NNarray.txt", "r");
    l    = fopen("data/locs.txt", "r");

    double y[100*100];
    double locs[100*100*2];
    int NNarray[100*100*81];
    

    for(int  i = 0; i < (100*100); ++i){
      fscanf(obs, "%lf", &y[i]);
    } 

    for(int i = 0; i < (100*100*2); ++i){
      fscanf(l, "%lf", &locs[i]); 
    }
    
    for(int i = 0; i < (100*100*81); ++i){
      fscanf(nn, "%d", &NNarray[i]);
    }

    
    fclose(obs);
    fclose(nn);
    fclose(l);


    double ll = 0.0;
    double covparms[4] = {4,0.1,0.5,0.1};
    int n = 100*100;
    int dim  = 2;
    int m = 81;
    int n_cores = 4;
   
   
   double t1 = omp_get_wtime();

    vecchia_likelihood(&ll,
		       covparms, 
		       y,
		       n,
		       locs,
		       dim,
		       NNarray,
		       m,
		       n_cores);
  
   double t2 = omp_get_wtime();


//   double t3 = omp_get_wtime();
//   int mb = n*(n+1)/2;
//   double* covmat   = (double*) malloc(mb*sizeof(double));
//   exponential_isotropic(covmat, m, covparms, locs, dim);
//   double t4 = omp_get_wtime();

   printf("likelihood:%lf\nlikelihood time:%lf\n", ll, t2-t1);
   
}

File: time.R
library(Rcpp)
#sourceCpp("~/GpGp/src/RcppExports")
sourceCpp("~/GpGp/src/matrix_mult.cpp", verbose = F)
sourceCpp("~/GpGp/src/vecchia_loglik_grad_info.cpp", verbose = F)

source("~/GpGp/R/nearest_neighbor_functions.R")
source("~/GpGp/R/ordering_functions.R")
source("~/GpGp/R/fit_model.R")
source("~/GpGp/R/simulation_functions.R")
source("nn_function.R")

gsize <- 100
nvec <- c(gsize,gsize)
x1 <- (1:nvec[1])/nvec[1]
x2 <- (1:nvec[2])/nvec[2]
locs <- as.matrix(expand.grid(x1,x2))
write.table(as.vector(locs), "data/locs.txt", row.names = F, col.names = F)

covparms <- c(variance = 4, range = 0.1, smoothness = 0.5, nugget = 0.1)

print("simulating data...")
y <- fast_Gp_sim(covparms[c(1,2,4)], "exponential_isotropic",locs,20)
write.table(y,"data/y.txt", row.names = F, col.names = F)

m <- 80

print("finding nearest neighbors...")
NNarray <- find_ordered_nn_brute(locs, m)
NNarray2 <- find_ordered_nn_brute2(locs, m)
NNarray2[is.na(NNarray2)] = 0
write.table(as.vector(t(NNarray2)),"data/NNarray.txt", row.names = F, col.names = F)

#print("Computing likelihoods...")
t1 = Sys.time()
l <- system.time(c <- vecchia_meanzero_loglik(covparms, "exponential_isotropic", y, locs, NNarray))
t2 = Sys.time()
t2-t1
l
c
#l1 <- system.time(exponential_isotropic(covparms, locs[1:31,]))
#l1
